autoload/ag.vim

```diff
@@ -58,6 +58,112 @@ function! ag#AgBuffer(cmd, args)
   call ag#Ag(a:cmd, a:args . ' ' . join(l:files, ' '))
 endfunction
 
+function! ag#AgGroup(ncontext, args)
+  silent! wincmd P
+  if !&previewwindow
+    execute 'bo ' . &previewheight . ' new'
+    set previewwindow
+  endif
+
+  setlocal modifiable
+
+  execute "silent %delete_"
+  
+  setlocal buftype=nofile bufhidden=wipe noswapfile nobuflisted nowrap
+
+  let context = ''
+  if a:ncontext > 0
+    let context = '-C' . a:ncontext 
+  endif
+
+  let searchstr = substitute(a:args,'#','\\#','g')
+  let searchstr = substitute(searchstr,'%','\\%','g')
+  execute 'silent read !ag --group --column ' . context . ' '. searchstr 
+  syn match agLine /^\d\+:\d\+\(:\)\@=/
+  syn match agLineContext /^\d\+-/
+  syn match agFile /^\n.\+$/hs=s+1
+  if hlexists('agSearch')
+    silent syn clear agSearch
+  endif
+  if a:args =~ '^"'
+    "detect "find me" file1 file2
+    let searchstr = split(a:args, '"')[0]
+  else
+    let searchstr = split(a:args, '\s\+')[0]
+  endif
+  let searchstr = substitute(searchstr, '/', '\\/', 'g')
+  execute 'syn match agSearch /' . searchstr . '/'
+  highlight link agLine LineNr
+  highlight link agFile Question
+  highlight link agSearch Todo
+  highlight link agLineContext Constant
+  setlocal foldmethod=expr
+  setlocal foldexpr=FoldAg()
+  setlocal foldcolumn=2
+  1
+  setlocal nomodifiable
+  map <buffer> o za
```

Comment:
Tbh I would also expect this buffer to have the same mappings as the other buffer's created by ag.vim. It's kinda confusing that open vertical split and  the like are not available, and that o doesn't do the same in this buffer than in the others. What do you think? Would it be a good idea to reuse the existing mappings in this command?